### PR TITLE
[DCA] Better handling of the envvar & doc

### PR DESCRIPTION
**Pull request by CharlyF**
### What does this PR do?

nests the `dd_cluster_agent_enabled` to have:

```
cluster_agent:
  enabled: true
  auth_token: foobar
```

### Motivation

Easier deployment.

### CODE REVIEW

File: Dockerfiles/cluster-agent/README.md [hunk 1]
```diff
@@ -8,12 +8,13 @@ The following environment variables are supported:
 
 - `DD_API_KEY` - **required** - your [Datadog API key][https://app.datadoghq.com/account/settings#api].
 - `DD_HOSTNAME`: hostname to use for the DCA.
-- `DD_CLUSTER_AGENT_CMD_PORT`: Port for the DCA to serve, default is `5005`.
-- `DD_USE_METADATA_MAPPER`: Enables the cluster level metadata mapping, default is `true`.
+- `DD_CLUSTER_AGENT_CMD_PORT`: port for the DCA to serve, default is `5005`.
+- `DD_USE_METADATA_MAPPER`: enables the cluster level metadata mapping, default is `true`.
```
Comment (devonboyer) [inline Dockerfiles/cluster-agent/README.md]: I'm confused, if this `README` is specific to the DCA, how come we have agent config options in here like `DD_USE_METADATA_MAPPER`?

**Review by devonboyer — COMMENTED** [review]
Can you also change `enable_hpa` to `cluster_agent.hpa.enabled`?

The env var would become `DD_CLUSTER_AGENT_HPA_ENABLED`.

File: pkg/config/config.go [hunk 2]
```diff
@@ -212,7 +212,7 @@ func init() {
 	Datadog.SetDefault("kube_resources_namespace", "")
 
 	// Datadog cluster agent
-	Datadog.SetDefault("cluster_agent", false)
+	Datadog.SetDefault("cluster_agent.enabled", false)
 	Datadog.SetDefault("cluster_agent.auth_token", "")
 	Datadog.SetDefault("cluster_agent.url", "")
 	Datadog.SetDefault("cluster_agent.kubernetes_service_name", "dca")
```
Comment (devonboyer) [inline pkg/config/config.go]: We also discussed changing the default service name from `dca` to `datadog-cluster-agent`. This would require updates to other docs and the sample manifests.

**Comment by CharlyF** [comment]
@DataDog/container-integrations - I spotted that the go routine `c.apiClient.StartMetadataMapping()` was called in the detect method of the kubernetes collector of the tagger, therefore in the node agent. This method should **only** be used by the Cluster Agent.
This is unnecessary as we do a `processKubeServices ` when calling `addToCacheMetadataMapping` (in the Pull and the Fetch) which uses the list of pod collected from the kubelet, thus local, and lists the Endpoints from the API server.

I am leaving the `kubernetes_apiserver_poll_freq` as it is meant to be used by the Cluster Agent. It will have no effect on the Node Agent. On the other hand, the `kubernetes_metadata_tag_update_freq` will not change, it is the frequency at which the Node Agent fetches cluster level metadata (from the APIserver or the Cluster Agent).

The benefits: 
- Backward compatible, and we fix a bug
- Can separate the 2 different processes (pulling from the DCA, pulling from the APIServer).

The drawback:
- Confusing

Thoughts ?

File: .gitignore [hunk 3]
```diff
@@ -44,3 +44,6 @@ datadog-agent*_amd64.deb
 /test/e2e/scripts/setup-instance/id_rsa.pub
 /test/e2e/scripts/setup-instance/ignition.json
 /test/e2e/containers/fake_datadog/venv/
+
+# Cluster Agent's config file
+Dockerfiles/cluster-agent/datadog-cluster.yaml
```
Comment (mfpierre) [inline .gitignore]: Why is this added to the gitignore? it seems the file was already tracked

File: pkg/tagger/collectors/kubernetes_main.go [hunk 4]
```diff
@@ -47,20 +47,18 @@ func (c *KubeMetadataCollector) Detect(out chan<- []*TagInfo) (CollectionMode, e
...
 			log.Errorf("Could not initialise the communication with the DCA, falling back to local service mapping: %s", errDCA.Error())
 		}
 	}
-	if !config.Datadog.GetBool("cluster_agent") || errDCA != nil {
+	if !config.Datadog.GetBool("cluster_agent.enabled") || errDCA != nil {
 		c.apiClient, err = apiserver.GetAPIClient()
 		if err != nil {
 			return NoCollection, err
 		}
-		c.apiClient.StartMetadataMapping()
```
Comment (mfpierre) [inline pkg/tagger/collectors/kubernetes_main.go]: if you remove this call you can remove the whole block

Comment (CharlyF) [reply pkg/tagger/collectors/kubernetes_main.go]: I think it is legit to leave the rest of the block as if we can't talk to the API Server, we don't want to start the `kube-metadata-collector` (so the Detect will fail)

File: .gitignore (hunk 3 above)
Comment (CharlyF) [reply .gitignore]: yes, I was planning on moving it in a different spot - I added this here as the `inv cluster-agent.builld` creates a datadog-cluster.yaml at this location. I think `invoke` should not alter the git state. I'll work on that in a different PR.

**Review by mfpierre — DISMISSED** [review]
LGTM ✅

File: pkg/tagger/collectors/kubernetes_main.go [hunk 5]
```diff
@@ -47,20 +47,18 @@ func (c *KubeMetadataCollector) Detect(out chan<- []*TagInfo) (CollectionMode, e
 		return NoCollection, err
 	}
 	// if no DCA or can't communicate with the DCA run the local service mapper.
```
Comment (mfpierre) [inline pkg/tagger/collectors/kubernetes_main.go]: is this comment still relevant?

Comment (mfpierre) [reply pkg/tagger/collectors/kubernetes_main.go]: @CharlyF ok I see it's called in the `Pull` I was a bit confused, thought it was something to start like the DCA one

**Review by mfpierre — APPROVED** [review]
